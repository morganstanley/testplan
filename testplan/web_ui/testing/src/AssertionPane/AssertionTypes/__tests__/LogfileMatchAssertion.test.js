import React from "react";
import { render } from "@testing-library/react";
import { StyleSheetTestUtils } from "aphrodite";

import LogfileMatchAssertion from "../LogfileMatchAssertion";

function defaultProps() {
  return {
    assertion: {
      utc_time: "2020-01-01T00:00:00.000000+00:00",
      machine_time: "2020-01-01T00:00:00.000000+00:00",
      type: "LogfileMatch",
      meta_type: "assertion",
      category: "DEFAULT",
      passed: false,
      description: null,
      timeout: 2.0,
      results: [
        {
          matched: "okok",
          pattern: ".*ok.*",
          start_pos: "<BOF>",
          end_pos: "<inode 110000, position 5>",
        },
        {
          matched: "okok",
          pattern: ".*ok.*",
          start_pos: "<inode 110000, position 5>",
          end_pos: "<inode 110000, position 10>",
        },
      ],
      failure: [
        {
          matched: null,
          pattern: ".*ok.*",
          start_pos: "<inode 110000, position 10>",
          end_pos: "<inode 110000, position 15>",
        },
      ],
    },
  };
}

describe("LogfileMatchAssertion", () => {
  let props;
  let component;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    component = undefined;
  });

  it("renders the correct HTML structure", () => {
    component = render(<LogfileMatchAssertion {...props} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
