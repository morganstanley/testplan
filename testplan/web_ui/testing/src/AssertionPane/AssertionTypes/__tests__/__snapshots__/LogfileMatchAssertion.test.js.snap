// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogfileMatchAssertion renders the correct HTML structure 1`] = `
<DocumentFragment>
  <div>
    <span>
      <b>
        Failed in 2 seconds.
      </b>
    </span>
  </div>
  <div>
    <span
      class="infoText_1m1rj1t"
    >
      Match between &lt;BOF&gt; and &lt;inode 110000, position 5&gt; found.
    </span>
  </div>
  <div>
    <strong
      class="title_81sjt7"
    >
      Pattern:
    </strong>
    <span
      style="color: black; background: rgb(245, 242, 240); text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 0px; margin: .5em 0px; overflow: auto;"
    >
      <code
        class="language-regex"
        style="color: black; background: none; text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
      >
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
        <span>
          ok
        </span>
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
      </code>
    </span>
  </div>
  <div>
    <strong
      class="title_81sjt7"
    >
      Log Line:
    </strong>
    <span
      class="inputText_197f7f1"
    >
      okok
    </span>
  </div>
  <div>
    <span
      class="infoText_1m1rj1t"
    >
      Match between &lt;inode 110000, position 5&gt; and &lt;inode 110000, position 10&gt; found.
    </span>
  </div>
  <div>
    <strong
      class="title_81sjt7"
    >
      Pattern:
    </strong>
    <span
      style="color: black; background: rgb(245, 242, 240); text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 0px; margin: .5em 0px; overflow: auto;"
    >
      <code
        class="language-regex"
        style="color: black; background: none; text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
      >
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
        <span>
          ok
        </span>
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
      </code>
    </span>
  </div>
  <div>
    <strong
      class="title_81sjt7"
    >
      Log Line:
    </strong>
    <span
      class="inputText_197f7f1"
    >
      okok
    </span>
  </div>
  <div>
    <span
      class="infoText_1m1rj1t"
    >
      No match from &lt;inode 110000, position 10&gt; found, search ended at &lt;inode 110000, position 15&gt;
    </span>
  </div>
  <div>
    <strong
      class="title_81sjt7"
    >
      Pattern:
    </strong>
    <span
      style="color: black; background: rgb(245, 242, 240); text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 0px; margin: .5em 0px; overflow: auto;"
    >
      <code
        class="language-regex"
        style="color: black; background: none; text-shadow: 0 1px white; font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace; font-size: 1em; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; word-wrap: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
      >
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
        <span>
          ok
        </span>
        <span
          class="token char-set"
          style="color: rgb(221, 74, 104);"
        >
          .
        </span>
        <span
          class="token quantifier"
          style="color: rgb(153, 0, 85);"
        >
          *
        </span>
      </code>
    </span>
  </div>
</DocumentFragment>
`;
